src/app/Controllers/MoviesController.cs

```diff
@@ -30,64 +32,51 @@ public MoviesController(ILogger<MoviesController> logger, IDAL dal)
         /// <summary>
         /// Returns a JSON array of Movie objects
         /// </summary>
-        /// <param name="q">(optional) The term used to search by movie title (rings)</param>
-        /// <param name="genre">(optional) Movies of a genre (Action)</param>
-        /// <param name="year">(optional) Get movies by year (2005)</param>
-        /// <param name="rating">(optional) Get movies with a rating >= rating (8.5)</param>
-        /// <param name="actorId">(optional) Get movies by Actor Id (nm0000704)</param>
-        /// <param name="pageNumber">1 based page index</param>
-        /// <param name="pageSize">page size (1000 max)</param>
+        /// <param name="movieQueryParameters"></param>
         [HttpGet]
-        public async Task<IActionResult> GetMoviesAsync([FromQuery] string q = null, [FromQuery] string genre = null, [FromQuery] int year = 0, [FromQuery] double rating = 0, [FromQuery] string actorId = null, [FromQuery] int pageNumber = 1, [FromQuery] int pageSize = Constants.DefaultPageSize)
+        public async Task<IActionResult> GetMoviesAsync([FromQuery] MovieQueryParameters movieQueryParameters)
         {
-            string method = GetMethodText(q, genre, year, rating, actorId, pageNumber, pageSize);
-
-            // validate query string parameters
-            var result = ParameterValidator.Movies(HttpContext?.Request?.Query, q, genre, year, rating, actorId, pageNumber, pageSize, method, logger);
-            if (result != null)
-            {
-                return result;
-            }
+            _ = movieQueryParameters ?? throw new ArgumentNullException(nameof(movieQueryParameters));
 
             // convert to zero based page index
-            pageNumber = pageNumber > 1 ? pageNumber - 1 : 0;
-
-            return await ResultHandler.Handle(dal.GetMoviesAsync(q, genre, year, rating, actorId, pageNumber * pageSize, pageSize), method, Constants.MoviesControllerException, logger).ConfigureAwait(false);
+            movieQueryParameters.PageNumber = movieQueryParameters.PageNumber > 1 ? movieQueryParameters.PageNumber - 1 : 0;
+
+            return await ResultHandler.Handle(dal.GetMoviesAsync(
```

Comment:
why not pass the class?

Reply:
see comment on the actors controller. this has cascading impacts which will increase the amount of change in the PR and has no linkage to the original issue.